### PR TITLE
PYSCAN-44 Read extra analysis properties from pyproject.toml files

### PR DESCRIPTION
**Pull request by guillaume-dequenne-sonarsource**

### CODE REVIEW

File: src/py_sonar_scanner/configuration.py [hunk 1]
```diff
@@ -60,6 +60,9 @@ def _read_wrapper_arguments(self):
         argument_parser.add_argument("-toml.path", "-Dtoml.path", "--toml.path", dest="toml_path")
         argument_parser.add_argument("-project.home", "-Dproject.home", "--project.home", dest="project_home")
         argument_parser.add_argument("-X", action="store_true", dest="debug")
+        argument_parser.add_argument(
+            "-read.toml", "-Dread.toml", "--read.toml", "-read_toml", action="store_true", dest="read_toml"
```
Comment (guillaume-dequenne-sonarsource) [inline src/py_sonar_scanner/configuration.py:64]: Not sure what the name should actually be, this is placeholder.

Also unsure whether the default value should be true or false.

While `tool.sonar` properties will have precedence over this, I'm mostly afraid that properties deduced from seemingly random fields would have precedence over the content of a `sonar-project.properties` file (and I feel like parsing those to check for duplicate properties might be overengineered).

File: src/py_sonar_scanner/configuration.py [hunk 2]
```diff
@@ -83,6 +90,23 @@ def _extract_sonar_properties(self, toml_properties):
         sonar_properties = tool_data["sonar"]
         return sonar_properties if isinstance(sonar_properties, dict) else {}
 
+    def _extract_common_properties(self, toml_properties):
+        properties = {}
+        if "tool" in toml_properties.keys() and "poetry" in toml_properties["tool"]:
```
Comment (guillaume-dequenne-sonarsource) [inline src/py_sonar_scanner/configuration.py:95]: If we decide to move forward with this approach, I'd create tickets to cover other tools (like Hatch) and reframe this one as a Poetry-centered one as a baby step.

**Review by joke1196 — APPROVED** [review]
Looks good to me! I would maybe just add a small bit of documentation for this feature in the readme

File: src/py_sonar_scanner/configuration.py (hunk 1 above)
Comment (guillaume-dequenne-sonarsource) [reply src/py_sonar_scanner/configuration.py]: I renamed to `read_project_config` which I guess may be a bit better, but I'm still not convinced.

**Pull request merged**